### PR TITLE
SSH Client rekey updates

### PR DESCRIPTION
**Pull request by redhatrises**
#### Description:

- Move to appropriate group
- Allow either ssh_config or ssh_config.d based of security preference
- Readability updates

### CODE REVIEW

File: linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/rule.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+ocil: |-
+    To verify that <tt>RekeyLimit</tt> is set correctly, run the following command:
+    <pre>$ sudo grep RekeyLimit /etc/ssh/ssh_config /etc/ssh/ssh_config.d/*.conf</pre>
+    If configured properly, output should contain something similar to
+    <pre>RekeyLimit {{{ sub_var_value("var_ssh_client_rekey_limit_size") }}} {{{
+    sub_var_value("var_ssh_client_rekey_limit_time") }}}</pre>
+
+warnings:
+    functionality: |-
```
Comment (vojtapolasek) [inline linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/rule.yml:44]:
```suggestion
    - functionality: |-
```

File: linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/rule.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        When using the <tt>ssh_client</tt> include configuration files,
+        ensure that there is no other <tt>RekeyLimit</tt> configuration preceding
+        the <tt>include</tt> directive in the <tt>/etc/ssh/ssh_config</tt>
+        configuration file.
+    functionality: |-
```
Comment (vojtapolasek) [inline linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/rule.yml:49]:
```suggestion
    - functionality: |-
```

File: linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/oval/shared.xml [hunk 3]
```diff
@@ -4,11 +4,13 @@
     <metadata>
       <title>{{{ rule_title }}}</title>
       {{{- oval_affected(products) }}}
-      <description>Ensure 'RekeyLimit' is configured with the correct value in /etc/ssh/ssh_config and /etc/ssh/ssh_config.d/*.conf</description>
+      <description>Ensure 'RekeyLimit' is configured with the correct value in /etc/ssh/ssh_config or /etc/ssh/ssh_config.d/*.conf</description>
     </metadata>
-    <criteria comment="RekeyLimit is correctly configured for ssh client" operator="AND">
-      <criterion comment="check that RekeyLimit is not configured in /etc/ssh/ssh_config" test_ref="test_ssh_client_rekey_limit_main_config" negate="true" />
-      <criterion comment="check correct RekeyLimit configuration in /etc/ssh/ssh_config.d/*.conf" test_ref="test_ssh_client_rekey_limit_include_configs" />
+    <criteria comment="RekeyLimit is correctly configured for ssh client" operator="OR">
```
Comment (vojtapolasek) [inline linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/oval/shared.xml:9]:
```suggestion
    <criteria comment="RekeyLimit is correctly configured for ssh client in the main configuration file" operator="OR">
```

File: linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/oval/shared.xml [hunk 4]
```diff
@@ -4,11 +4,13 @@
...
+      <criterion comment="check correct RekeyLimit is configured in /etc/ssh/ssh_config" test_ref="test_ssh_client_rekey_limit_main_config" />
+      <criteria comment="RekeyLimit is correctly configured for ssh client" operator="AND">
```
Comment (vojtapolasek) [inline linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/oval/shared.xml:11]:
```suggestion
      <criteria comment="RekeyLimit is correctly configured for ssh client in included configuration file" operator="AND">
```

File: linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/oval/shared.xml [hunk 5]
```diff
@@ -4,11 +4,13 @@
...
+        <criterion comment="check /etc/ssh/ssh_config.d/ is configured in /etc/ssh/ssh_config" test_ref="test_ssh_client_rekey_limit_include" />
+        <criterion comment="check correct RekeyLimit configuration in /etc/ssh/ssh_config.d/*.conf" test_ref="test_ssh_client_rekey_limit_include_configs" />
     </criteria>
```
Comment (vojtapolasek) [inline linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/oval/shared.xml:14]: one closing `</criteria>` tag missing, prevents build

**Comment by mildas** [comment]
**Changes identified:**
Rule ssh_client_rekey_limit:
 Deleted attribute from OVAL check.
 Node moved within OVAL check.
 New node inserted to OVAL check.
 Text changed in OVAL check.
 Attribute value changed in OVAL check.

**Recommended tests to execute:**
 build_product rhel8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml ssh_client_rekey_limit

**Comment by mildas** [comment]
**Changes identified:**
Rule ssh_client_rekey_limit:
 Attribute value changed in OVAL check.
 Node moved within OVAL check.
 Deleted attribute from OVAL check.
 New node inserted to OVAL check.
 Text changed in OVAL check.
Rule configure_crypto_policy:
 Found change in bash remediation.

**Recommended tests to execute:**
 build_product ol8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml configure_crypto_policy
 build_product rhel8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml ssh_client_rekey_limit

File: linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/oval/shared.xml (hunk 3 above)
Comment (redhatrises) [reply linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/oval/shared.xml:9]: Actually, the first comment is correct as RekeyLimit is configured for sshd client OR in an includes file.

File: linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/rule.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+description: |-
+    The <tt>RekeyLimit</tt> parameter specifies how often
+    the session key is renegotiated, both in terms of
+    amount of data that may be transmitted and the time
+    elapsed. To decrease the default limits, configure the
+    <tt>RekeyLimit</tt> in <tt>/etc/ssh/ssh_config</tt>:
+    <pre>RekeyLimit {{{ sub_var_value("var_ssh_client_rekey_limit_size") }}} {{{ sub_var_value("var_ssh_client_rekey_limit_time") }}}</pre>
+
+    Alternatively, an <tt>ssh_config</tt> includes file can be created
+    to configure <tt>RekeyLimit</tt>. To configure <tt>RekeyLimit</tt>
```
[the remark this answers is not in the document]
Comment (redhatrises) [reply linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/rule.yml:16]: Handling the include directive itself is really needed in a new rule as it is a new configuration item. Also, it is configured out of the box by default. However, a new warning could be added for this. Also, now that I think about it, the OVAL should also handle some of the warnings as well.

**Comment by mildas** [comment]
**Changes identified:**
Rule ssh_client_rekey_limit:
 New node inserted to OVAL check.
 Node moved within OVAL check.
 Deleted attribute from OVAL check.
 Text changed in OVAL check.
 Attribute value changed in OVAL check.

**Recommended tests to execute:**
 build_product rhel8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml ssh_client_rekey_limit

File: linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/oval/shared.xml (hunk 3 above)
Comment (vojtapolasek) [reply linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/oval/shared.xml:9]: ah you are right.

**Comment by vojtapolasek** [comment]
Thank you for fixes. Unfortunately this test is still failing:
bad_main_config_good_include_config.fail.sh 
Also please remove the old *.var files from system/software/integrity/crypto because the build fails.
I suggest you also add tests which will verify the newly added criteria, e.g. the include directive is there but the included file has wrong value or vice versa.

**Comment by mildas** [comment]
**Changes identified:**
Rule ssh_client_rekey_limit:
 Node moved within OVAL check.
 Attribute value changed in OVAL check.
 Deleted attribute from OVAL check.
 Text changed in OVAL check.
 New node inserted to OVAL check.

**Recommended tests to execute:**
 build_product rhel8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml ssh_client_rekey_limit

File: linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/rule.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (vojtapolasek) [reply linux_os/guide/services/ssh/ssh_client/ssh_client_rekey_limit/rule.yml:16]: OK. You can add it as a warning. What warnings should OVAL check for? Does it even make sense to check for warnings? I believe that warnings are there as a mean to say to the customer that there are certain cases which can't be handled by the content but they should be aware of.

**Comment by redhatrises** [comment]
> Thank you for fixes. Unfortunately this test is still failing:
> bad_main_config_good_include_config.fail.sh
> Also please remove the old *.var files from system/software/integrity/crypto because the build fails.
> I suggest you also add tests which will verify the newly added criteria, e.g. the include directive is there but the included file has wrong value or vice versa.

@vojtapolasek Tests haven't been updated yet hence the issues that you are seeing.

**Comment by matejak** [comment]
test this please

**Comment by mildas** [comment]
**Changes identified:**
Rule ssh_client_rekey_limit:
 Text changed in OVAL check.
 Attribute value changed in OVAL check.
 New node inserted to OVAL check.
 Deleted attribute from OVAL check.
 Node moved within OVAL check.

**Recommended tests to execute:**
 build_product rhel8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-rhel8-ds.xml ssh_client_rekey_limit

**Comment by matejak** [comment]
There is also a typo: `'docucmentation_complete': 'true'` docu**c**mentation

**Comment by matejak** [comment]
test this please

**Comment by redhatrises** [comment]
Closing this for now as the changes that need to be made won't be for a while.

**Pull request closed without merging**